esp32_s3_korvo_2/esp32_s3_korvo_2.c

```diff
@@ -0,0 +1,596 @@
+    BSP_NULL_CHECK_GOTO(buf2, ERR);
+    // initialize LVGL draw buffers
+    lv_disp_draw_buf_init(&disp_buf, buf1, buf2, BSP_LCD_H_RES * 50);
+
+    ESP_LOGD(TAG, "Register display driver to LVGL");
+    lv_disp_drv_init(&disp_drv);
+    disp_drv.hor_res = BSP_LCD_H_RES;
+    disp_drv.ver_res = BSP_LCD_V_RES;
+    disp_drv.flush_cb = lvgl_port_flush_callback;
+    disp_drv.drv_update_cb = lvgl_port_update_callback;
+    disp_drv.draw_buf = &disp_buf;
+    disp_drv.user_data = panel_handle;
+    return lv_disp_drv_register(&disp_drv);
+
+#if (!CONFIG_BSP_ERROR_CHECK)
+ERR:
+    if (buf1) {
+        free(buf1);
+    }
+    if (buf2) {
+        free(buf2);
+    }
+
+    return NULL;
+#endif
+}
+
+static esp_err_t lvgl_port_indev_init(void)
+{
+    static lv_indev_drv_t indev_drv_tp;
+    lv_indev_t *indev_touchpad;
+
+    /* Initialize touch */
+    const esp_lcd_touch_config_t tp_cfg = {
+        .x_max = BSP_LCD_H_RES,
+        .y_max = BSP_LCD_V_RES,
+        .rst_gpio_num = GPIO_NUM_NC, // Shared with LCD reset
+        .int_gpio_num = BSP_LCD_TOUCH_INT,
+        .levels = {
+            .reset = 0,
+            .interrupt = 0,
+        },
+        .flags = {
+            .swap_xy = 0,
+            .mirror_x = 1,
+            .mirror_y = 0,
+        },
+    };
+    esp_lcd_panel_io_handle_t tp_io_handle = NULL;
+    const esp_lcd_panel_io_i2c_config_t tp_io_config = ESP_LCD_TOUCH_IO_I2C_TT21100_CONFIG();
+    BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_new_panel_io_i2c((esp_lcd_i2c_bus_handle_t)BSP_I2C_NUM, &tp_io_config, &tp_io_handle));
+    BSP_ERROR_CHECK_RETURN_ERR(esp_lcd_touch_new_i2c_tt21100(tp_io_handle, &tp_cfg, &tp));
+    assert(tp);
+
+    /* Register a touchpad input device */
+    lv_indev_drv_init(&indev_drv_tp);
+    indev_drv_tp.type = LV_INDEV_TYPE_POINTER;
+    indev_drv_tp.read_cb = bsp_touchpad_read;
+    indev_drv_tp.user_data = tp;
+    indev_touchpad = lv_indev_drv_register(&indev_drv_tp);
+    BSP_NULL_CHECK(indev_touchpad, ESP_ERR_NO_MEM);
+
+    return ESP_OK;
+}
+#endif //BSP_DISPLAY_ENABLED
+
+esp_err_t bsp_display_brightness_set(int brightness_percent)
+{
+    ESP_LOGW(TAG, "This board doesn't support to change brightness of LCD");
+    return ESP_ERR_NOT_SUPPORTED;
+}
+
+esp_err_t bsp_display_backlight_off(void)
+{
+    return bsp_display_brightness_set(0);
+}
+
+esp_err_t bsp_display_backlight_on(void)
+{
+    return bsp_display_brightness_set(100);
+}
+
+lv_disp_t *bsp_display_start(void)
+{
+#if BSP_DISPLAY_ENABLED
+    lv_init();
+    lv_disp_t *disp = lvgl_port_display_init();
+    BSP_ERROR_CHECK_RETURN_NULL(lvgl_port_indev_init());
+    BSP_ERROR_CHECK_RETURN_NULL(lvgl_port_tick_init());
+    lvgl_mux = xSemaphoreCreateMutex();
+    BSP_NULL_CHECK(lvgl_mux, NULL);
+    xTaskCreate(lvgl_port_task, "LVGL task", 4096, NULL, CONFIG_BSP_DISPLAY_LVGL_TASK_PRIORITY, NULL);
+    return disp;
+#else
+    ESP_LOGW(TAG, "Display is disabled, because it is not tested yet!");
+    return NULL;
+#endif
+}
```

Comment:
Could we use the new `esp_lvgl_port` component?

Reply:
Yes, but we haven't got HW for the LCD. I prepared before the component was released and disabled it. I didn't want to remove it. Is it ok to change it to lvgl_port, after we will have the LCD HW?